### DISCUSSION TITLE
What are the main impacts and issues when using Linkerd but all meshed services are using the same "default" ServiceAccount?

### DISCUSSION THREAD

**Discussion by Aracki** [linkerd/linkerd2#12019] [Help] [answered]
I cannot understand this from the documentation. 

Apart from the security concerns and wrong names for `client_id` in metrics, would Linkerd work seamlessly if all meshed pods are utilizing the same "default" ServiceAccount? Would it be there some issues regarding rotating secrets for mTLS?

**Comment by alpeb** [comment, marked as answer]
Things should work fine if you use the same SA all over the place, including mTLS. But it's good practice to have different SAs per workload; identity in linkerd is tied to SAs, so in order to apply some policy to a workload you need to refer to its SA, so it needs to be differentiated.
